aggregator/handler/config.go

```diff
@@ -22,56 +22,72 @@ package handler
 
 import (
 	"errors"
+	"fmt"
+	"sort"
 	"time"
 
 	"github.com/m3db/m3aggregator/aggregator"
 	"github.com/m3db/m3x/instrument"
 	"github.com/m3db/m3x/log"
 	"github.com/m3db/m3x/retry"
+
+	"github.com/uber-go/tally"
 )
 
 var (
 	errUnknownFlushHandlerType         = errors.New("unknown flush handler type")
 	errNoForwardHandlerConfiguration   = errors.New("no forward handler configuration")
+	errNoShardedHandlerConfiguration   = errors.New("no sharded handler configuration")
 	errNoBroadcastHandlerConfiguration = errors.New("no broadcast handler configuration")
 )
 
 // FlushHandlerConfiguration contains configuration for flushing metrics.
 type FlushHandlerConfiguration struct {
 	// Flushing handler type.
-	Type string `yaml:"type" validate:"regexp=(^blackhole$|^logging$|^forward$|^broadcast$)"`
+	Type Type `yaml:"type"`
 
 	// Forward handler configuration.
 	Forward *forwardHandlerConfiguration `yaml:"forward"`
 
+	// Sharded handler configuration.
+	Sharded *shardedHandlerConfiguration `yaml:"sharded"`
+
 	// Forward handler configuration.
```

Comment:
nit: should this be "Broadcast handler configuration"?

Reply:
No longer applies